### ISSUE TITLE
Confirm modal: Exclamation mark out of place

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/server#15778]
From: https://github.com/nextcloud/server/pull/15774#issuecomment-496557238

![](https://user-images.githubusercontent.com/5836855/58489078-d285a080-816a-11e9-91de-cc0bfe3f00f2.png)

> Looks a bit blurry to me, and ! is out of place?

@ChristophWurst
> yes, but that is an issue with the confirm modal in general. mind opening a ticket for that and pinging our designers? thx

@nextcloud/designers

**Comment by skjnldsv** [comment]
> > Looks a bit blurry to me, and ! is out of place?

Your window screen have an impair height. So the translate is antialiasing on the subpixel :)
We should indeed try to center vertically the popup differently :)

**Comment by jancborchardt** [comment]
The icon itself is out of place here, it does not need to be there.

Additionally the buttons should not say "Yes" and "No" but be more descriptive like "Wipe data from this device" and "Keep data".

**Comment by georgehrke** [comment]
This is actually not that trivial. Definitely no good first issue.

`OC.Dialogs.confirm`, etc. only allows you either an OK button or YES/NO. It doesn't allow custom labels.
cc @ChristophWurst Something we should do when porting it to vue.

**Comment by pierlon** [comment]
`OC.Dialogs.confirm` is a wrapper for `OC.Dialogs.message`, so calling `OC.Dialogs.message` with the custom buttons could be a workaround for now.

**Comment by georgehrke** [comment]
@pierlon Please check the code please :)
You can only set an OK button or YES/NO. Even when using `OC.Dialogs.message` directly

**Comment by pierlon** [comment]
Yep you're right, overlooked that. Perhaps you could try changing the text after its created:

```js
OC.dialogs.confirm('Text', 'Title', null, true).done(() => {
    const buttons = $('.oc-dialog-buttonrow button')

    buttons.eq(0).text('Keep data')
    buttons.eq(1).text('Wipe data')
})

**Comment by georgehrke** [comment]
Possible yes, but quite hacky and definitely nothing we should teach new contributors 😉
There are plans to have a clean, no-jquery implementation in our [vue-components](https://github.com/nextcloud/nextcloud-vue). We should just make sure to allow custom button labels there and migrate all usages of `OC.Dialogs` to our vue component.

**Comment by szaimen** [comment]
Hi, please update to 24.0.8 or better 25.0.2 and report back if it fixes the issue. Thank you!
